test/Constraints/pack-expansion-expressions.swift

```diff
@@ -397,3 +397,14 @@ do {
   _ = Defaulted(t: "b", 0) // Ok
   _ = Defaulted(t: "c", 1.0, u: "d", 0) // Ok
 }
+
+// rdar://108064941 - unused result diagnostic is unaware of Void packs
+func test_no_unused_result_warning(arr: inout [Any]) {
+  func test1<each T>(_ value: (repeat each T)) {
+    repeat arr.append(each value.element) // no warning
+  }
+
+  func test2<each T>(_ value: repeat each T) {
+    ((repeat arr.append(each value))) // no warning
```

Comment:
Is there any reason to do this with a tuple instead of just writing the underlying pack expansion expression? I'm wondering if we should still diagnose this for tuples

Reply:
I did it that way because users don't actually know that this is a tuple, syntactically it's just parens which seems reasonable to make consistent with non-variadic cases.